test/visual/backgrounds.js

```diff
@@ -0,0 +1,72 @@
+const test = require('tape');
+const fs = require('fs');
+const PNG = require('pngjs').PNG;
+const pixelmatch = require('pixelmatch');
+const createBackgroundScreenshot = require('./helpers/createBackgroundScreenshot');
+
+const fixturePath = 'test/visual/fixtures/';
+const tempDir = fs.mkdtempSync(fixturePath);
+
+const readPNG = (pngPath) => new Promise(resolve => fs.createReadStream(pngPath).pipe(new PNG()).on('parsed', resolve));
+
+async function createScreenshot(effectName) {
+  await createBackgroundScreenshot(effectName, tempDir);
+
+  // If there is no accepted image for this background (ex: it's a new background),
+  // use this screenshot as accepted image
+  if (!fs.existsSync(`${fixturePath}${effectName}.png`)) {
+    await createBackgroundScreenshot(effectName, fixturePath);
+  }
+}
+
+async function testBackground(t, effect) {
+  await createScreenshot(effect);
+
+  const [actual, expected] = await Promise.all([
+    readPNG(`${tempDir}/${effect}.png`),
+    readPNG(`${fixturePath}${effect}.png`)
+  ]);
+
+  const diff = new PNG({width: actual.width, height: actual.height});
+
+  let pixelDiff = pixelmatch(actual.data, expected.data, diff.data, actual.width, actual.height, {threshold: 0.1});
+  t.equal(pixelDiff, 0, effect);
+}
+
+[
+  'none',
+  'swirl',
+  'rainbow',
+  'color_cycle',
+  'disco',
+  'diamonds',
+  'strobe',
+  'rain',
+  'text',
+  'raining_tacos',
+  'splatter',
+  'spiral',
+  'spotlight',
+  'color_lights',
+  'snowflakes'
+].forEach(effect => {
```

Comment:
Great for now - even better would be to automatically extract the set of backgrounds from `Effect.js` somehow... :grin: